Added support for latest iOS Splash-Screens

juliantroeps opened the pull request:
- iPhone 8 Plus
- iPhone 7 Plus
- iPhone 6s Plus
- iPhone 6 Plus
- iPhone X
- iPhone Xs
- iPhone Xr 
- iPhone Xs Max
- iPad 2019 (10.7")

evilebottnawi commented:
Can you provide link on docs about this icons?

juliantroeps commented:
The sizes and resolutions are available at the Human Interface Guidelines from Apple:
https://developer.apple.com/design/human-interface-guidelines/ios/icons-and-images/launch-screen/
https://developer.apple.com/design/human-interface-guidelines/ios/icons-and-images/image-size-and-resolution/

Additionally made edits based on this guidelines:
https://medium.com/appscope/adding-custom-ios-splash-screens-to-your-progressive-web-app-41a9b18bdca3

danielg2002 commented:
hi all, any update around this PR ? should be very useful.

thanks,
Daniel.

danielg2002 commented:
@juliantroeps 

landcape -> land**s**cape

    ({ relative }) => `<link rel="apple-touch-startup-image" media="(device-width: 414px) and (device-height: 736px) and (orientation: landcape) and (-webkit-device-pixel-ratio: 3)" href="${relative("apple-touch-startup-image-2208x1242.png")}"> `,

danielg2002 commented:
@juliantroeps 

duplicated line!

({ relative }) => `<link rel="apple-touch-startup-image" media="(device-width: 768px) and (device-height: 1024px) and (orientation: portrait) and (-webkit-device-pixel-ratio: 2)" href="${relative("apple-touch-startup-image-1536x2008.png")}">`,

juliantroeps commented:
Cannot get the tests to pass. Will look into this tomorrow.

juliantroeps commented:
Both checks now passed @evilebottnawi

orangeloops commented:
@juliantroeps I found some bugs related to resolution/orientation/sizes...

for example:

"apple-touch-startup-image-2208x1242.png": {
	      **"width": 2208,
	      "height": 1242,**
	      "transparent": false,
	      **"rotate": true,**
	      "mask": false
	    },

VS

"apple-touch-startup-image-2436x1125.png": {
	      **"width": 1125,
	      "height": 2436,**
	      "transparent": false,
	      **"rotate": true,**
	      "mask": false
	    },

both are rotated, but width / height have a different criteria...

for that reason I create a new full html.js and icons.json without any rotation, I will share it in a couple of minutes

orangeloops commented:
@juliantroeps @evilebottnawi 
I did a full update using this table:

    // Device              Portrait size      Landscape size     Screen size        Pixel ratio
    // iPhone SE            640px × 1136px    1136px ×  640px     320px ×  568px    2
    // iPhone 8             750px × 1334px    1334px ×  750px     375px ×  667px    2
    // iPhone 7             750px × 1334px    1334px ×  750px     375px ×  667px    2
    // iPhone 6s            750px × 1334px    1334px ×  750px     375px ×  667px    2
    // iPhone XR            828px × 1792px    1792px ×  828px     414px ×  896px    2
    // iPhone XS           1125px × 2436px    2436px × 1125px     375px ×  812px    3
    // iPhone X            1125px × 2436px    2436px × 1125px     375px ×  812px    3
    // iPhone 8 Plus       1242px × 2208px    2208px × 1242px     414px ×  736px    3
    // iPhone 7 Plus       1242px × 2208px    2208px × 1242px     414px ×  736px    3
    // iPhone 6s Plus      1242px × 2208px    2208px × 1242px     414px ×  736px    3
    // iPhone XS Max       1242px × 2688px    2688px × 1242px     414px ×  896px    3
    // 9.7" iPad           1536px × 2048px    2048px × 1536px     768px × 1024px    2
    // 7.9" iPad mini 4    1536px × 2048px    2048px × 1536px     768px × 1024px    2
    // 10.5" iPad Pro      1668px × 2224px    2224px × 1668px     834px × 1112px    2
    // 11" iPad Pro        1668px × 2388px    2388px × 1668px     834px × 1194px    2
    // 12.9" iPad Pro      2048px × 2732px    2732px × 2048px    1024px × 1366px    2

Source: https://stackoverflow.com/a/57338818

Please, compare attached files:
[Archive.zip](https://github.com/itgalaxy/favicons/files/3631037/Archive.zip)

juliantroeps commented:
@orangeloops @evilebottnawi Thanks for the reply. I shouldn't create PRs if I am not 100% available haha.

Added your changes to the commit. Tests and Icons look good. All generated sizes match all current iOS screen sizes. I added support for new 10.2" base iPad.

evilebottnawi commented:
/cc @brunocodutra

juliantroeps commented:
any updates on this? @evilebottnawi @brunocodutra

evilebottnawi commented:
Looks @brunocodutra do not have enough time, i am too, we need maintainer for project, sorry

brunocodutra commented:
@juliantroeps really sorry to keep you waiting, you've put a lot of effort in this and more than deserved a timely review. Many thanks to @orangeloops and @danielg2002 for stepping in and help reviewing. 

You guys rock and clearly care a lot about `favicons`, have you consider joining us as maintainers?

brunocodutra commented on test/createExamples.js:
```diff
@@ -0,0 +1,76 @@
+    Run `node createExamples.js` to create the icons, slashs and files.
+**/
+
+var favicons = require('../dist/index.js'),
```
Any particular reason why using `var` here as opposed to `const`?

juliantroeps replied:
Indeed, it's 2019.

brunocodutra commented on test/createExamples.js:
```diff
@@ -0,0 +1,76 @@
+            yandex: true                // Create Yandex browser icon. `boolean` or `{ offset, background, mask, overlayGlow, overlayShadow }`
+        }
+    },
+    callback = function (error, response) {
```
Not super important, but I'd expect an arrow function here instead of the more verbose `function` syntax.

juliantroeps replied:
Updated the callback function to the arrow syntax. Left the error callback in the fswriteFile methods untouched for now. 
The file is just to manually create sample files on the fly and has no deeper meaning.

brunocodutra commented on test/createExamples.js:
```diff
@@ -0,0 +1,76 @@
+
+        // Save images
+        response.images.forEach(item => {
+            fs.writeFile(__dirname + '/tmp/' + item.name, item.contents, 'binary', function(err){
```
Have you considered using string interpolation?

brunocodutra commented on test/createExamples.js:
```diff
@@ -0,0 +1,76 @@
+    callback = function (error, response) {
+        if (error) {
+            console.log(error.message); // Error description e.g. "An unknown error has occurred"
+            return;
```
Should we `throw` here instead?

brunocodutra commented on test/createExamples.js:
```diff
@@ -0,0 +1,76 @@
+
+        // Save files
+        response.files.forEach(item => {
+            fs.writeFile(__dirname + '/tmp/' + item.name, item.contents, 'binary', function(err){
```
I'd also favor string interpolation here and pretty much everywhere else, where strings are explicitly concatenated by the `+` operator.

brunocodutra commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "favicons",
-  "version": "5.4.1",
+  "version": "5.4.2",
```
I'd argue this should rather be a minor bump to `5.5.0`, thoughts?

danielg2002 commented:
@brunocodutra @juliantroeps hi all, we also found additional bugs related to Windows icons...

pls, review attached screen

![image](https://user-images.githubusercontent.com/11050021/66435197-ad4b4d80-e9fa-11e9-925e-0b0e6ee6cc98.png)

danielg2002 commented:
Attached our latest icons.json definition...
[icons.json.zip](https://github.com/itgalaxy/favicons/files/3704614/icons.json.zip)

juliantroeps commented:
@brunocodutra Sure. I agree with all your comments. That configuration is based on the example in the readme. Should we keep it like that or update it as well? 

@danielg2002 Thanks. Good catch. I'll update the config right away.

brunocodutra commented:
> Should we keep it like that or update it as well?

We should definitely update it, but I think it should go on a separate PR.
